### PR TITLE
Improve JP2KAK documentation

### PR DESCRIPTION
**Pull request by palemieux**

### CODE REVIEW

File: doc/source/drivers/raster/jp2kak.rst [hunk 1]
```diff
@@ -256,13 +261,13 @@ documentation to better understand their meaning.
    assigned roughly logarithmically spaced bit-rates. When only one rate is specified, an internal heuristic determines
    a lower bound and logarithmically spaces the layer rates over the range. The rates have to be in ASC order.
 
-Known Kakadu Issues
+Known Kakadu SDK Issues
```
Comment (rouault) [inline doc/source/drivers/raster/jp2kak.rst:264]: 4 missing dashes in following line

File: doc/source/drivers/raster/jp2kak.rst [hunk 2]
```diff
@@ -275,7 +280,7 @@ to version 7.9.
 kdu_get_num_processors always returns 0 for some platforms
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-On non-windows / non-mac installs (e.g. Linux), Kakadu might not include
+On non-windows / non-mac installs (e.g. Linux), Kakadu SDK ight not include
```
Comment (rouault) [inline doc/source/drivers/raster/jp2kak.rst:283]:
```suggestion
On non-windows / non-mac installs (e.g. Linux), Kakadu SDK might not include
```

**Comment by rouault** [comment]
```
/__w/gdal/gdal/doc/source/drivers/raster/jp2kak.rst:28: WARNING: Bullet list ends without a blank line; unexpected unindent.
/__w/gdal/gdal/doc/source/drivers/raster/jp2kak.rst:33: WARNING: Bullet list ends without a blank line; unexpected unindent.
/__w/gdal/gdal/doc/source/drivers/raster/jp2kak.rst:: ERROR: Anonymous hyperlink mismatch: 1 references but 0 targets.
```

**Comment by palemieux** [comment]
What is the easiest way to check RST syntax without a full build?

**Comment by rouault** [comment]
> What is the easiest way to check RST syntax without a full build?

There might perhaps be online RST checker ? Otherwise you can make a local build of the doc : https://gdal.org/development/dev_documentation.html

File: doc/source/drivers/raster/jp2kak.rst [hunk 3]
```diff
@@ -1,46 +1,50 @@
...
+
+* reading both JPEG 2000 codestreams (.j2c or .jpc) and JP2 files (.jp2). New
+  images can be also written, but existing images cannot be updated in place.
+
+* lossy and lossless compression of 8-bit and 16-bit images with 1 or more bands
+  (components).
+
+* GeoTIFF style coordinate system and georeferencing information in JP2 file via
+  the
+  `GeoJP2(tm)<https://web.archive.org/web/20151028081930/http://www.lizardtech.com/download/geo/geotiff_box.txt>`__
```
Comment (rouault) [inline doc/source/drivers/raster/jp2kak.rst:28]:
```suggestion
  `GeoJP2(tm) <https://web.archive.org/web/20151028081930/http://www.lizardtech.com/download/geo/geotiff_box.txt>`__
```

File: doc/source/drivers/raster/jp2kak.rst [hunk 4]
```diff
@@ -54,15 +58,14 @@ Driver capabilities
 Configuration Options
 ---------------------
 
-The JP2KAK driver supports the following `Config
-Options <http://trac.osgeo.org/gdal/wiki/ConfigOptions>`__. These
+The JP2KAK driver supports the following `Config Options <http://trac.osgeo.org/gdal/wiki/ConfigOptions>`__. These
```
Comment (rouault) [inline doc/source/drivers/raster/jp2kak.rst:61]:
```suggestion
The JP2KAK driver supports the following :ref:`configoptions`. These
```

**Pull request merged**